nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -158,34 +172,28 @@ in
     };
 
     users.extraGroups = mkIf (cfg.group == "ipfs") {
-      ipfs = {
-        gid = config.ids.gids.ipfs;
-      };
+      ipfs.gid = config.ids.gids.ipfs;
     };
 
-    systemd.services.ipfs-init = {
+    systemd.services.ipfs-init = recursiveUpdate commonEnv {
       description = "IPFS Initializer";
 
       after = [ "local-fs.target" ];
       before = [ "ipfs.service" "ipfs-offline.service" ];
 
-      environment.IPFS_PATH = cfg.dataDir;
-
-      path  = [ pkgs.ipfs pkgs.su pkgs.bash ];
-
       preStart = ''
         install -m 0755 -o ${cfg.user} -g ${cfg.group} -d ${cfg.dataDir}
       '';
       script = ''
         if [[ ! -f ${cfg.dataDir}/config ]]; then
-          ${ipfs}/bin/ipfs init ${optionalString cfg.emptyRepo "-e"}
+          ipfs init ${optionalString cfg.emptyRepo "-e"}
         fi
-        ${ipfs}/bin/ipfs --local config Addresses.API ${cfg.apiAddress}
-        ${ipfs}/bin/ipfs --local config Addresses.Gateway ${cfg.gatewayAddress}
+        ipfs --local config Addresses.API ${cfg.apiAddress}
+        ipfs --local config Addresses.Gateway ${cfg.gatewayAddress}
       '' + optionalString cfg.autoMount ''
-        ${ipfs}/bin/ipfs --local config Mounts.FuseAllowOther --json true
-        mkdir -p $(${ipfs}/bin/ipfs --local config Mounts.IPFS)
-        mkdir -p $(${ipfs}/bin/ipfs --local config Mounts.IPNS)
+        ipfs --local config Mounts.FuseAllowOther --json true
+        mkdir -p $(ipfs --local config Mounts.IPFS)
+        mkdir -p $(ipfs --local config Mounts.IPNS)
```

Comment:
This should be in `preStart` as normal users like `ipfs` are not allowed to create the directories `/ipfs` `/ipns`.
However ipfs is not initialized in `preStart`, so `$(ipfs --local config Mounts.IPFS)` is not possible. Maybe make the mount points configurable in nix and then use `cfg.ipfsMountPoint` and set pass that to ipfs accordingly.

Reply:
Come to think of it, that stuff shouldn't be part of the -init service anyway. I'll move it to the preStart of the other 3 daemon services.